packages/jellyfish-core/src/methods/mining.ts

```diff
@@ -0,0 +1,23 @@
+import {Client} from '../api'
+
+export interface MintingInfo {
+  blocks: number
+  difficulty: number
+  isoperator: boolean
+  networkhashps: number
+  pooledtx: number
+  chain: 'main' | 'test' | 'regtest'
```

Comment:
I suggest if we can have an enum/const for this network variable. I guess it will also be widely used on apps.

Reply:
Ultimately the `methods/*.ts` and their interfaces is to allow high separation of concern between rpc categories. There are a few benefits for not putting it into a const/enum.

1. enum sucks in an agnostic language ideology, it's basically a integer/string construct with added salt
2. code coupling, unnecessary coupling of variables that can be separated. someone wants to change something but this is owned by 100x code owner. 
3. poor developer experience, especially for declaring an interface. It's should be self contained. Check out the interface once without the need to travel down 10x files to find something
4. TypeScript already support interoperability/compatibility so user can declare their enum ontop

Reply:
Make sure to support `custom` too for privately-run chain. We also have an official `devnet` that's more for private internal testing use. https://github.com/DeFiCh/ain/blob/c5e2338eb540c8623bf5725284de1ef4eb67b898/src/chainparamsbase.cpp#L16

Reply:
Should be ok for me. It's more of preference I think. Main idea is to reuse especially these commonly used terms.